### PR TITLE
check that we are connected before calling ray commands

### PR DESCRIPTION
**Pull request by robertnishihara**

### CODE REVIEW

File: doc/install-on-macosx.md [hunk 1]
```diff
@@ -34,8 +34,8 @@ source setup-env.sh # Add Ray to your Python path.
 ```
 
 For convenience, you may also want to add the line `source
-"$RAY_ROOT/setup-env.sh"` to your `~/.bashrc` file manually, where `$RAY_ROOT`
-is the Ray directory (e.g., `/home/ubuntu/ray`).
+"$RAY_ROOT/setup-env.sh"` to the bottom of your `~/.bashrc` file manually, where
```
Comment (mehrdadn) [inline doc/install-on-macosx.md]: Does it matter if it's the bottom or the top? (Why?)

File: lib/python/ray/worker.py [hunk 2]
```diff
@@ -437,6 +437,15 @@ def submit_task(self, func_name, args):
...
 """
 
+def check_connected(worker=global_worker):
+  """Check if the worker is connected.
+
+  Raises:
+    Exception: An exception is raised if the worker is not connected.
+  """
+  if worker.handle is None:
+    raise Exception("This command cannot be called before a Ray cluster has not been started. You can start one with 'ray.services.start_ray_local(num_workers=1)'")
```
Comment (mehrdadn) [inline lib/python/ray/worker.py]: "before a Ray cluster has not been started"?

File: doc/install-on-macosx.md (hunk 1 above)
Comment (pcmoritz) [inline doc/install-on-macosx.md]: It doesn't, but typically programs (and people) add this kind of stuff at the bottom. There is already lots of commands in the file, and the ones at the top are typically used by the system.

File: lib/python/ray/worker.py (hunk 2 above)
Comment (robertnishihara) [inline lib/python/ray/worker.py]: Fixed.

**Pull request merged**